skyxplore-ui-test/src/test/java/com/mentoring/test/EquipmentTest.java

```diff
@@ -1,61 +1,126 @@
                 .clickCex01ProductionButton();
-
         assertThat(factoryPage.isExtenderItemInQueueVisible())
-                .as("Extender item in queue should be visible.")
+                .as("Extender item in Queue should be visible.")
                 .isTrue();
-        assertThat(factoryPage.getQuantityInQueue().equals(DEFAULT_QUANTITY))
-                .as("Quantity in queue is not correct")
+        softAssertion.assertThat(factoryPage.getQuantityInQueue().equals(DEFAULT_QUANTITY))
+                .as("Quantity in Queue should be correct.")
                 .isTrue();
-        assertThat(factoryPage.isQueueProcessStarted())
-                .as("Queue process is not started.")
+        softAssertion.assertThat(factoryPage.isQueueProcessBarVisible())
+                .as("Queue process bar should be visible.")
                 .isTrue();
 
-        //TODO: Subtask-3: Production time of CEX-01 is updated and item is finished
-        //TODO: Subtask-4: CEX-01 item is added to the ship
+        log.info("Updating DB: Set queue process end-time to finish the production.");
+        factoryId = DbSelects.getFactoryId(database, user.getEmail(), characterName);
+        DbUpdates.setProductionEndTimeByFactoryId(database, factoryId, "0");
 
-        //softAssertion.assertAll();
-    }
+        driver.navigate().refresh();
+        assertThat(factoryPage.isQueueProcessBarVisible())
+                .as("Queue process bar should be updated")
+                .isTrue();
+        assertThat(factoryPage.isQueueDisappeared())
+                .as("Queue should be disappeared")
+                .isTrue();
+
+        EquipmentPage equipPage = new FactoryPage(driver)
+                .openOverviewPage()
+                .openEquipmentPage();
+        assertThat(equipPage.isStorageCex01ItemVisible())
+                .as("CEX-01 item should be visible in Storage section")
+                .isTrue();
+        assertThat(equipPage.isShipBat01ItemVisible())
+                .as("BAT-01 item should be visible in Ship section")
+                .isTrue();
 
+        equipPage.removeShipBat01Item();
+        assertThat(equipPage.isShipEmptySlotVisible())
+                .as("After removing BAT-01 item, an Empty slot should be visible in Ship section")
+                .isTrue();
+        softAssertion.assertThat(equipPage.isStorageBat01ItemVisible())
+                .as("After moving BAT-01 item from Ship section, it should be visible in Storage section")
+                .isTrue();
+
+        equipPage.moveStorageCex01ItemToShipEmptySlot();
+        assertThat(equipPage.isShipCex01ItemVisible())
+                .as("After moving CEX-01 from Storage section, it should be visible in Ship section")
+                .isTrue();
+        assertThat(equipPage.getShipEmptySlots() == equipPage.getShipCex01EmptySlotAttribute())
+                .as("CEX-01 empty slots attribute and empty slots in Ship section should be the same.")
+                .isTrue();
```

Comment:
These could be soft asserts (BAT-01 won't be used and there won't be further steps after drag n' drop)

Reply:
Ok, fixed, thx.